services/py-api/src/service/participants_service.py

```diff
@@ -35,10 +38,29 @@ async def _create_participant_and_team_in_transaction(
             return participant
 
         return Ok((participant.ok_value, team.ok_value))
+    
+    async def _check_capacity(self) -> bool:
+        """Calculate if there is enough capacity to register a new team"""
+        # Fetch number of verified random participants
+        verified_random_participants = await self._participant_repo.get_verified_random_participants_count()
+
+        # Fetch number of verified registered teams
+        verified_registered_teams = await self._team_repo.get_verified_registered_teams_count()
+
+        # Calculate the anticipated number of teams
+        number_ant_teams = verified_registered_teams + math.ceil(verified_random_participants / self.team_capacity)
+
+        # Check against the hackathon capacity
+        return number_ant_teams < self.hackathon_cap
```

Comment:
I think this method should be in the TeamsRepository and it should be public. As we are checking if we are able to add a Team, we are counting teams. And it should be `check_hackathon_capacity`